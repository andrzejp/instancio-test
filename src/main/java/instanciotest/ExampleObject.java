package instanciotest;

import java.time.LocalDateTime;
import java.util.List;

public record ExampleObject(String aString, LocalDateTime aDateTime, Integer anInteger, List<Long> aListOfLongs) {
}
