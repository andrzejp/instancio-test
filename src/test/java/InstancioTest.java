import instanciotest.ExampleObject;
import org.approvaltests.Approvals;
import org.instancio.Instancio;
import org.junit.jupiter.api.Test;

class InstancioTest {

    @Test
    void should_instantiate_an_object_with_random_values() {
        ExampleObject exampleObject = Instancio.create(ExampleObject.class);
        Approvals.verify(exampleObject);
    }
}
