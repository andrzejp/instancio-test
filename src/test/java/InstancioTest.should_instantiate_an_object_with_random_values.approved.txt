ExampleObject[aString=OMTTHYHV, aDateTime=2003-04-12T05:36:32.179609726, anInteger=2473, aListOfLongs=[638, 1750]]